transport/grpc/headers.go

```diff
@@ -153,9 +168,21 @@ func metadataToTransportRequest(md metadata.MD) (*transport.Request, error) {
 				request.Encoding = transport.Encoding(getContentSubtype(value))
 			}
 		default:
+			if isReservedHeaderPrefixV2(header) {
+				reportStrippedHeader = true
```

Comment:
instead of a true/false value, do you want to use a counter here? Right now, no matter how many violations we have, we already inc by 1.